rust/datafusion/src/physical_plan/mod.rs

```diff
@@ -80,8 +80,9 @@ pub trait ExecutionPlan: Debug + Send + Sync {
         children: Vec<Arc<dyn ExecutionPlan>>,
     ) -> Result<Arc<dyn ExecutionPlan>>;
 
-    /// creates an iterator
-    async fn execute(&self, partition: usize) -> Result<SendableRecordBatchStream>;
+    /// creates a vector of streams of [RecordBatch]es, each stream corresponding
+    /// to a part of the partition.
+    async fn execute(&self) -> Result<Vec<SendableRecordBatchStream>>;
```

Comment:
I am sorry I have just started looking at this PR more carefully -- it seems like the proposal is to change execute so that it returns a `Vec` of streams rather than a single stream. I think this is reasonable as it allows physical plan implementations more flexibility when reading the inputs (they can always merge the streams into one if they choose too)